Source/Core/VideoCommon/TextureCacheBase.cpp

```diff
@@ -615,13 +631,13 @@ class ArbitraryMipmapDetector
             const int diff_squared = diff * diff;
             pixel_diff += diff_squared;
           }
-          average_diff += pixel_diff;
+          current_diff_sum += pixel_diff;
         }
         ptr1 += shape.row_length;
         ptr2 += shape.row_length;
       }
 
-      return average_diff / (shape.width * shape.height * 4) / 2.56f;
+      return (float)current_diff_sum / (shape.width * shape.height * 4) / 2.56f;
```

Comment:
(Original code mistake) I don't trust this 2.56f constant. I believe this is trying to renormalize from the 256 range back into a percentage (0-100), but it doesn't realize it's adding up diff_squared here, so it should really be dividing by (65536/100.0f).

Reply:
Have tested and confirmed this fixes the erroneous mipmap detection issues.

Reply:
Yeah, that needs changing - it's not been 'correct' since my first tweak to square the difference. But I don't really understand what a "percentage difference" means - especially as we're explicitly scaling by diff^2 in the belief that large differences are more visually obvious even if they affect fewer pixels.

Reply:
I mean, it's not "percentage difference" anymore, it can just be a score from 1-100. We need to pick *some* maximum number, and 25,599 doesn't seem like it should be it.

Reply:
Same line, but unrelated to the discussion up there: `static_cast<float>`?

Reply:
Perhaps instead of changing the normalization, wrap the part before the normalization with a sqrt. According to @degasus this is the proper way to compare the result of mean squared error, which is basically what we're doing here.